manifests/git_repo.pp

```diff
@@ -0,0 +1,22 @@
+# Setup and configure a git repo
+define katello_devel::git_repo(
+  $source,
+  $github_username = undef,
+) {
+
+  validate_string($source)
+
+  $sources = {origin => "https://github.com/${source}.git"}
```

Comment:
We can push for convention through this :) If a more common idiom is to have username and upstream then I am all for that. Some retraining will have to occur but that's fine in my opinion.